### ISSUE TITLE
installation

### ISSUE THREAD

**Issue by swvanderlaan** [tidyverse/stringr#172]
Hi,
I've been trying for some time now, to install `stringr` and its dependency `stringi`. But whatever I do, via R-Studio or R, re-installing command line tools on my Mac, etc. It doesn't work. I get the error message below. 
Do you have any advice or solution?

Best,

Sander

`install.packages("stringr")`
`Installing package into ‘/usr/local/lib/R/3.4/site-library’`
`(as ‘lib’ is unspecified)`
`--- Please select a CRAN mirror for use in this session ---`
`Secure CRAN mirrors`
``
` 1: 0-Cloud [https]                 2: Algeria [https]`
` 3: Australia (Canberra) [https]    4: Australia (Melbourne) [https]`
` 5: Australia (Perth) [https]       6: Austria [https]`
` 7: Belgium (Ghent) [https]         8: Brazil (RJ) [https]`
` 9: Brazil (SP 1) [https]          10: Bulgaria [https]`
`11: Chile 1 [https]                12: China (Lanzhou) [https]`
`13: Colombia (Cali) [https]        14: Czech Republic [https]`
`15: Denmark [https]                16: Estonia [https]`
`17: France (Lyon 1) [https]        18: France (Lyon 2) [https]`
`19: France (Marseille) [https]     20: France (Montpellier) [https]`
`21: France (Paris 2) [https]       22: Germany (Göttingen) [https]`
`23: Germany (Münster) [https]      24: Greece [https]`
`25: Iceland [https]                26: Indonesia (Jakarta) [https]`
`27: Ireland [https]                28: Italy (Padua) [https]`
`29: Japan (Tokyo) [https]          30: Malaysia [https]`
`31: Mexico (Mexico City) [https]   32: Norway [https]`
`33: Philippines [https]            34: Serbia [https]`
`35: Spain (A Coruña) [https]       36: Spain (Madrid) [https]`
`37: Sweden [https]                 38: Switzerland [https]`
`39: Turkey (Denizli) [https]       40: Turkey (Mersin) [https]`
`41: UK (Bristol) [https]           42: UK (Cambridge) [https]`
`43: UK (London 1) [https]          44: USA (CA 1) [https]`
`45: USA (IA) [https]               46: USA (KS) [https]`
`47: USA (MI 1) [https]             48: USA (OR) [https]`
`49: USA (TN) [https]               50: USA (TX 1) [https]`
`51: Vietnam [https]                52: (other mirrors)`
``
``
`Selection: 52`
`Other CRAN mirrors`
``
` 1: 0-Cloud                       2: Algeria`
` 3: Argentina (La Plata)          4: Australia (Canberra)`
` 5: Austria                       6: Belgium (Antwerp)`
` 7: Belgium (Ghent)               8: Brazil (BA)`
` 9: Brazil (PR) [https]          10: Brazil (PR)`
`11: Brazil (RJ)                  12: Brazil (SP 1)`
`13: Brazil (SP 2)                14: Bulgaria`
`15: Canada (BC)                  16: Canada (MB) [https]`
`17: Canada (MB)                  18: Canada (NS)`
`19: Canada (ON)                  20: Chile 1`
`21: Chile 2 [https]              22: China (Beijing) [https]`
`23: China (Beijing)              24: China (Hefei) [https]`
`25: China (Hefei)                26: China (Lanzhou)`
`27: China (Xiamen)               28: Colombia (Cali)`
`29: Czech Republic               30: Denmark`
`31: Ecuador                      32: El Salvador`
`33: Estonia                      34: France (Lyon 2)`
`35: France (Marseille)           36: France (Montpellier)`
`37: France (Paris 1)             38: France (Paris 2)`
`39: Germany (Göttingen)          40: Germany (Münster)`
`41: Greece                       42: Hungary`
`43: Iceland                      44: India [https]`
`45: India                        46: Iran`
`47: Ireland                      48: Italy (Milano)`
`49: Italy (Padua)                50: Italy (Palermo)`
`51: Japan (Tokyo)                52: Korea (Seoul 1)`
`53: Korea (Seoul 2)              54: Korea (Ulsan)`
`55: Malaysia                     56: Mexico (Mexico City)`
`57: Mexico (Texcoco)             58: Netherlands (Amsterdam)`
`59: Netherlands (Utrecht)        60: New Zealand [https]`
`61: New Zealand                  62: Norway`
`63: Philippines                  64: Portugal (Lisbon)`
`65: Portugal (Porto)             66: Singapore`
`67: South Africa (Cape Town)     68: South Africa (Johannesburg)`
`69: Spain (A Coruña)             70: Spain (Madrid)`
`71: Sweden                       72: Switzerland`
`73: Taiwan (Chungli) [https]     74: Taiwan (Chungli)`
`75: Taiwan (Taipei)              76: Thailand`
`77: Turkey (Denizli)             78: Turkey (Mersin)`
`79: UK (Bristol)                 80: UK (Cambridge)`
`81: UK (London 1)                82: UK (London 2)`
`83: USA (CA 1)                   84: USA (CA 2)`
`85: USA (IA)                     86: USA (IN) [https]`
`87: USA (IN)                     88: USA (KS)`
`89: USA (MI 1)                   90: USA (MO)`
`91: USA (NC)                     92: USA (OH 1)`
`93: USA (OH 2)                   94: USA (OR)`
`95: USA (PA 1)                   96: USA (PA 2)`
`97: USA (TN)                     98: USA (TX 1)`
`99: Venezuela`
``
`Selection: 59`
`also installing the dependency ‘stringi’`
``
`trying URL 'http://cran-mirror.cs.uu.nl/src/contrib/stringi_1.1.5.tar.gz'`
`Content type 'application/x-gzip' length 3645872 bytes (3.5 MB)`
`==================================================`
`downloaded 3.5 MB`
``
`trying URL 'http://cran-mirror.cs.uu.nl/src/contrib/stringr_1.2.0.tar.gz'`
`Content type 'application/x-gzip' length 94095 bytes (91 KB)`
`==================================================`
`downloaded 91 KB`
``
`* installing *source* package ‘stringi’ ...`
`** package ‘stringi’ successfully unpacked and MD5 sums checked`
`checking for local ICUDT_DIR... icu55/data`
`checking for R_HOME... /usr/local/Cellar/r/3.4.0_1/R.framework/Resources`
`checking for R... /usr/local/Cellar/r/3.4.0_1/R.framework/Resources/bin/R`
`checking for R >= 3.1.0 for C++11 use... yes`
`checking for R < 3.4.0 for CXX1X flag use... no`
`checking for cat... /bin/cat`
`checking for gcc... /usr/local/opt/llvm/bin/clang -fopenmp`
`checking whether the C compiler works... no`
`configure: error: in /private/var/folders/xk/dfj20kgd56xg8796xh874tjr0000gn/T/RtmpPb4KSO/R.INSTALLdadf3d1a6883/stringi':`
`configure: error: C compiler cannot create executables`
`See config.log' for more details`
`ERROR: configuration failed for package ‘stringi’`
`* removing ‘/usr/local/lib/R/3.4/site-library/stringi’`
`ERROR: dependency ‘stringi’ is not available for package ‘stringr’`
`* removing ‘/usr/local/lib/R/3.4/site-library/stringr’`
` `
`The downloaded source packages are in ‘/private/var/folders/xk/dfj20kgd56xg8796xh874tjr0000gn/T/RtmpEXKxyC/downloaded_packages’`
`Warning messages:`
`1: In install.packages("stringr") :`
`installation of package ‘stringi’ had non-zero exit status`
`2: In install.packages("stringr") :`
`installation of package ‘stringr’ had non-zero exit status`

My `sessionInfo()`

`sessionInfo()`
`R version 3.4.0 (2017-04-21)`
`Platform: x86_64-apple-darwin16.6.0 (64-bit)`
`Running under: macOS Sierra 10.12.5`
` `
`Matrix products: default`
`BLAS/LAPACK: /usr/local/Cellar/openblas/0.2.19_1/lib/libopenblasp-r0.2.19.dylib`
` `
`locale:`
`[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8`
` `
`attached base packages:`
`[1] stats     graphics  grDevices utils     datasets  methods   base`
` `
`other attached packages:`
`[1] BiocInstaller_1.26.0`
` `
`loaded via a namespace (and not attached):`
`[1] compiler_3.4.0 tools_3.4.0`

**Comment by gagolews** [comment]
This has nothing to do with stringr. If needed, file an issue @ gagolews/stringi.
By the way, do you have xcode compiler collection installed?

**Comment by swvanderlaan** [comment]
I suspected as much. By "Xcode compiler collection", what do you mean? I've got Xcode installed and the command line tools. Do I need to install other stuff to?

**Comment by gagolews** [comment]
are you able to run `/usr/local/opt/llvm/bin/clang -fopenmp` ?

**Comment by gagolews** [comment]
Actually, the above message means that you do have a compiler installed. Are you able to compile other packages like Rcpp or dplyr?

**Comment by swvanderlaan** [comment]
Yes, I can install dplyr.

**Comment by swvanderlaan** [comment]
Mmm. I've further diagnosed the problem. It appears `git2r` can also not be installed, which is dependent on `zlib`. But the latter is installed - at least it should be, as I own a Mac (10.12.5). Do you have any idea on how to get R to "see" `zlib`?

**Comment by swvanderlaan** [comment]
Victory! I have discovered what I believe was the issue: a wrong path to the `xcode command line tools`.
`sudo xcode-select --reset` did the trick and now `xcode-select -p` points to `/Library/Developer/CommandLineTools`. And `which xcode-select` points to `/usr/bin/xcode-select`. Googling I discovered that it is _not_ needed to install Xcode through the App Store as it states on http://railsapps.github.io/xcode-command-line-tools.html : "On earlier versions of macOS, it was more difficult to install Xcode Command Line Tools. It required a huge download of the full Xcode package from the Mac App Store or registration as an Apple developer for a smaller Command Line Tools package. Mac OS X Mavericks made installation of Xcode Command Line Tools much simpler.".

Thanks!
